### ISSUE TITLE
Receive 2 or more streams in Python with different sampling rates

### ISSUE THREAD

**Issue by BCIstorm** [sccn/labstreaminglayer#283]
I'm trying to receive at least two streams with different sampling frequencies:
- EEG (8 channels) @ 250Hz;
- at least another one from an Eye-tracker (5 channels) @120fps.

What is the best way to do this in Python and save to a file for an offline analysis?

**Comment by cboulay** [comment]
You can save them to an xdf file with LabRecorder.

If  you also need real-time access in Python then you'll have to be more specific about what you want to do before we can offer advice on "the best way". i.e. real-time visualization is different than synchronizing the streams and finding EEG features that correlate with eye movements.

**Comment by BCIstorm** [comment]
Thank you @cboulay , I tryed LabRecorder (pre-compiled binary) and I guess it works (how do one knows if the sync is correct?). But then I need a xdf viewer (used SigViewer from https://github.com/cbrnr/sigviewer) or a Python script like this https://github.com/sccn/xdf
I would like to have as few dependencies as possible.

For now, I just want to synchronize the streams, save them to a file to do some post-processing.
But, eventually I'd like to have access to real-time visualization.

**Comment by dmedine** [comment]
Synchronization happens when the xdf data is loaded (either in Python or 
in Matlab) https://github.com/sccn/xdf.

Then, you can visualize your data using scipy or matlab. Online 
synchronization has to be done at the code level. When you create an 
inlet, you can give it flags that will tell it to perform online clock 
offset removal and de-jittering of the incoming data's timestamps. 
https://github.com/sccn/labstreaminglayer/blob/master/LSL/liblsl/include/lsl_cpp.h#L71-L82

**Comment by dmedine** [comment]
...and there is also a pretty good matlab based signal viewer 
https://github.com/sccn/labstreaminglayer/tree/master/Apps/MATLABViewer

**Comment by BCIstorm** [comment]
Thank you @dmedine , but I'm using Python, do not want to use Matlab now.

**Comment by BCIstorm** [comment]
@dmedine de-jittering is an average filter right? I want to do my own post-processing.

So, going to my first question, what's the best way, I mean, should I create an inlet for each stream and pull a sample from each one in alternate ways? Or pull a chunk? And what would be the the synchronization error?

**Comment by dmedine** [comment]
The de-jittering routine uses linear regression.

**Comment by Maryada-27** [comment]
How to use LabRecorder for ubuntu?

**Comment by cboulay** [comment]
Hi @Maryada-27 
Your question is unrelated to this issue. Please ask your question in a new issue, or you can even try to get one of us in real time in the Slack channel ([see README for link](https://github.com/sccn/labstreaminglayer#how-to-get-support)).

Be sure to include lots of information. What version of Ubuntu? What do you mean "use"? Are you trying to compile it, or is it already compiled and you just want to know how it works?
